StudyHub/StudyHub/Controllers/AssignmentController.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using StudyHub.BLL.Services.Interfaces;
+using StudyHub.Common.DTO.AssignmentDTO;
+
+namespace StudyHub.Controllers;
+
+[Route("[controller]")]
+[ApiController]
+public class AssignmentController : Controller
+{
+    private readonly IAssignmentService _assignmentService;
+
+    public AssignmentController(IAssignmentService assignmentService)
+    {
+        _assignmentService = assignmentService;
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> InsertAssignment(CreateAssignmentDTO dto)
+    {
+        var result = await _assignmentService.CreateAssignmentAsync(dto);
+        return Ok(result);
+    }
+
+    [HttpPut("{assignmentId}")]
+    public async Task<IActionResult> UpdateAssignment(Guid assignmentId, UpdateAssignmentDTO dto)
+    {
+        var result = await _assignmentService.UpdateAssignmentAsync(assignmentId, dto);
+        return Ok(result);
+    }
+
+    [HttpDelete("{id}")]
+    public async Task<IActionResult> DeleteAssignment(Guid id)
+    {
+        return await _assignmentService.DeleteAssignmentAsync(id) ? NoContent() : NotFound();
+    }
+
+    [HttpGet("{assignmentId}")]
+    public async Task<IActionResult> GetAssignment(Guid assignmentId)
+    {
+        var result = await _assignmentService.GetAssignmentByIdAsync(assignmentId);
+        return Ok(result);
+    }
+
+    [HttpGet("subject/{subjectId}")]
```

Comment:
I'm not sure, maybe better to have this endpoint in Subject controller

Coz `api/assignment/subject/{id}` looks missleading
As for me better looks `api/subject/{id}/assignments`